rpaas/consul_manager.py

```diff
@@ -147,6 +150,43 @@ def list_lua_modules(self, instance_name):
     def remove_lua(self, instance_name, lua_module_name, lua_module_type):
         self.write_lua(instance_name, lua_module_name, lua_module_type, None)
 
+    def add_server_upstream(self, instance_name, upstream_name, server):
+        upstream_name = self._host_from_destination(upstream_name)
+        server = self._host_from_destination(server)
+        servers = self._list_upstream(instance_name, upstream_name)
+        servers.add(server)
+        self._save_upstream(instance_name, upstream_name, servers)
+
+    def remove_server_upstream(self, instance_name, upstream_name, server):
+        upstream_name = self._host_from_destination(upstream_name)
+        server = self._host_from_destination(server)
+        servers = self._list_upstream(instance_name, upstream_name)
+        try:
```

Comment:
It's best to avoid using exceptions for flow control, this could be `if server in servers: servers.remove(server)`